### PR TITLE
Support for Debot Interfaces

### PR DESCRIPTION
**Pull request by Keshoid**
Allow to call virtual debot interfaces from debots.
1) filter interface calls from debot output messages (`run_output.rs`).
2) Add `send` callback to BrowserCallbacks.
3) Send interface requests  to Browser.
4) add `Dengine.send` method to receive responses from Browser.
5) Add test for interfaces to test debot browser.

### CODE REVIEW

File: ton_client/src/debot/dengine.rs [hunk 1]
```diff
@@ -729,6 +771,22 @@ impl DEngine {
         routines::call_routine(self.ton.clone(), name, args, signer).await
     }
 
+    async fn handle_output(&mut self, mut output: RunOutput) -> ClientResult<()> {
+        for msg in std::mem::take(&mut output.interface_calls) {
```
Comment (AlexeyVavilin) [inline ton_client/src/debot/dengine.rs:775]: You may use variable directly, without `std::mem::take`

File: ton_client/src/debot/dengine.rs [hunk 2]
```diff
@@ -729,6 +771,22 @@ impl DEngine {
...
+            // TODO: check if there are builtin interfaces
+            // BuiltinInterfaces::try_execute(&msg)
+            self.browser.send(msg.0).await;
+        }
+
+        // TODO: 
+        // result.send_msgs();
+
+        for _msg in std::mem::take(&mut output.get_method_calls) {
```
Comment (AlexeyVavilin) [inline ton_client/src/debot/dengine.rs:784]: You may use variable directly, without `std::mem::take`

File: ton_client/src/debot/dengine.rs [hunk 3]
```diff
@@ -179,6 +182,51 @@ impl DEngine {
         }
     }
 
+    pub async fn send(&mut self, source: String, func_id: u32, params: String) -> ClientResult<()> {
+        debug!("send from {} id = {} params = {}", source, func_id, params);
+        let params = serde_json::from_str(&params)
+            .map_err(|e| Error::invalid_json_params(e) )?;
+        let abi = Contract::load(self.raw_abi.as_bytes()).unwrap();
```
Comment (AlexeyVavilin) [inline ton_client/src/debot/dengine.rs:189]: No `unwrap`s are allowed in SDK. Check the result

File: ton_client/src/debot/dengine.rs [hunk 4]
```diff
@@ -179,6 +182,51 @@ impl DEngine {
...
+            ParamsOfRunTvm {
+                account: std::mem::take(&mut self.state),
+                message: msg_base64,
+                abi: Some(self.abi.clone()),
+                execution_options: None,
+            },
+        ).await?;
+        let mut run_output = RunOutput::new(
+            run_result.account,
+            run_result.decoded.unwrap().output,
```
Comment (AlexeyVavilin) [inline ton_client/src/debot/dengine.rs:223]: Is `decoded` field guaranteed to have proper value? Better check to be sure

**Review by AlexeyVavilin — CHANGES_REQUESTED** [review]
Fill changelog

File: ton_client/src/debot/dengine.rs (hunk 1 above)
Comment (Keshoid) [reply ton_client/src/debot/dengine.rs:775]: Fixed

File: ton_client/src/debot/dengine.rs (hunk 2 above)
Comment (Keshoid) [reply ton_client/src/debot/dengine.rs:784]: Fixed

File: ton_client/src/debot/dengine.rs (hunk 3 above)
Comment (Keshoid) [reply ton_client/src/debot/dengine.rs:189]: Fixed

File: ton_client/src/debot/dengine.rs (hunk 4 above)
Comment (Keshoid) [reply ton_client/src/debot/dengine.rs:223]: Fixed

**Review by AlexeyVavilin — APPROVED** [review]

**Pull request merged**